tracer/src/Datadog.Trace/Debugger/LiveDebugger.cs

```diff
@@ -286,18 +288,17 @@ internal void UpdateAddedProbeInstrumentations(IReadOnlyList<ProbeDefinition> ad
 
         private static void SetRateLimit(ProbeDefinition probe)
         {
-            if (probe is not LogProbe logProbe)
+            switch (probe)
             {
-                return;
-            }
-
-            if (logProbe.Sampling is { } sampling)
-            {
-                ProbeRateLimiter.Instance.SetRate(probe.Id, (int)sampling.SnapshotsPerSecond);
-            }
-            else
-            {
-                ProbeRateLimiter.Instance.SetRate(probe.Id, logProbe.CaptureSnapshot ? 1 : 5000);
+                case LogProbe { Sampling: { } sampling }:
+                    ProbeRateLimiter.Instance.SetRate(probe.Id, (int)sampling.SnapshotsPerSecond);
+                    break;
+                case LogProbe logProbe:
+                    ProbeRateLimiter.Instance.SetRate(probe.Id, logProbe.CaptureSnapshot ? 1 : 5000);
+                    break;
+                case SpanDecorationProbe or MetricProbe or SpanProbe:
```

Comment:
According to how `SetRateLimit` is being called, you won't enter into this branch with a `SpanProbe` since it's [being skipped](https://github.com/DataDog/dd-trace-dotnet/blob/master/tracer/src/Datadog.Trace/Debugger/LiveDebugger.cs#L253-L264). What more, I'm not sure it's even necessary for Span Probes - since there we have unique instrumentation where `ProbeProcessor` is not being used at all. Take a look at [SpanDebuggerInvoker](https://github.com/DataDog/dd-trace-dotnet/blob/master/tracer/src/Datadog.Trace/Debugger/Instrumentation/SpanDebuggerInvoker.cs) for reference.

Reply:
Thanks for noting that, fixed in 2b34914ca8d1bb9df2c6e7beb99b995e8ca525ab